File: app/src/main/java/com/example/ctap/fido2/ErrorResponse.java
package com.example.ctap.fido2;

import com.fasterxml.jackson.dataformat.cbor.CBORGenerator;

import java.io.IOException;

public class ErrorResponse extends Response {
    final static byte CTAP2_ERR_UNSUPPORTED_ALGORITHM = 0x26;
    final static byte CTAP2_ERR_OPERATION_DENIED = 0x27;

    ErrorResponse(final byte errorCode) {
        fidoStatus = errorCode;
    }

    @Override
    void serializeCBOR(CBORGenerator cborGenerator) throws IOException {
        // No-op.
    }
}

File: app/src/main/java/com/example/ctap/fido2/ErrorResponse.kt
package com.example.ctap.fido2

import com.example.ctap.ctap2.Response
import com.fasterxml.jackson.dataformat.cbor.CBORGenerator
import java.io.ByteArrayOutputStream

internal class ErrorResponse (errorCode: Byte)
    : Response(errorCode) {

    override fun serializeCBOR(gen: CBORGenerator,
                               output: ByteArrayOutputStream) {
        // No payload.
    }

    companion object {
        internal const val CTAP2_ERR_UNSUPPORTED_ALGORITHM: Byte = 0x26
        internal const val CTAP2_ERR_OPERATION_DENIED: Byte = 0x27
    }
}